openedx/core/lib/block_structure/transformers.py

```diff
@@ -0,0 +1,100 @@
+"""
+Module for a collection of BlockStructureTransformers.
+"""
+from logging import getLogger
+
+from .exceptions import TransformerException
+from .transformer_registry import TransformerRegistry
+
+
+logger = getLogger(__name__)  # pylint: disable=C0103
+
+
+class BlockStructureTransformers(object):
+    """
+    Class for a collection of block structure transformers.
+    """
+    def __init__(self, transformers=None, usage_info=None):
+        """
+        Arguments:
+            transformers ([BlockStructureTransformer]) - List of transformers
+                to add to the collection.
+
+            usage_info (any negotiated type) - A usage-specific object
+                that is passed to the block_structure and forwarded to all
+                requested Transformers in order to apply a
+                usage-specific transform. For example, an instance of
+                usage_info would contain a user object for which the
+                transform should be applied.
+
+        Raises:
+            TransformerException - if any transformer is not registered in the
+                Transformer Registry.
+        """
+        self.usage_info = usage_info
+        self.transformers = []
+        if transformers:
+            self.__iadd__(transformers)
+
+    def __iadd__(self, transformers):
```

Comment:
I had considered adding append and extend.  But does it make sense to do so even without having any existing callers?  Would you not consider that over-engineering?